src/blas_like/level3/Multiply.cpp

```diff
@@ -14,6 +14,60 @@
 namespace El {
 
 namespace {
+/**
+ * MultiplyCSR specialization where the CSR matrix happens to have all nonzeros = 1.
+ */
+template<typename T>
+void MultiplyCSR
+( Orientation orientation,
+  Int m, Int n,
+  T alpha,
+  const Int* rowOffsets,
+  const Int* colIndices,
+  const T*   x,
+  T beta,
+        T*   y )
+{
+    DEBUG_CSE
+    if( orientation == NORMAL )
+    {
+        for( Int i=0; i<m; ++i )
+        {
+            T sum = 0;
+            const Int eStart = rowOffsets[i];
+            const Int eStop = rowOffsets[i+1];
+            for( Int e=eStart; e<eStop; ++e )
+                sum += x[colIndices[e]];         
+            y[i] = alpha*sum + beta*y[i];
+        }
+    }
+    else
+    {
+        const bool conj = ( orientation == ADJOINT );
```

Comment:
There is no need to distinguish between `orientation == TRANSPOSE` and `orientation == ADJOINT` if the values are binary (or even if they are arbitrary real). It would be a good idea to delete the `if( conj )` branch.